jerry-core/ecma/builtin-objects/ecma-builtin-regexp.c

```diff
@@ -131,47 +162,66 @@ ecma_builtin_regexp_dispatch_construct (const ecma_value_t *arguments_list_p, /*
 #endif /* ENABLED (JERRY_ES2015) */
   }
 
-  ecma_string_t *pattern_string_p = ecma_regexp_read_pattern_str_helper (pattern_value);
-  ecma_value_t ret_value = ECMA_VALUE_ERROR;
+  ecma_object_t *new_target_obj_p = ecma_op_regexp_alloc (new_target_p);
 
-  if (pattern_string_p == NULL)
+  if (JERRY_UNLIKELY (new_target_obj_p == NULL))
   {
-    goto cleanup;
+    return ECMA_VALUE_ERROR;
   }
 
-  uint16_t flags = 0;
+  ecma_value_t ret_value;
 
-  if (!ecma_is_value_undefined (flags_value))
+#if ENABLED (JERRY_ES2015)
+  if (create_regexp_from_bc)
   {
-    ecma_string_t *flags_string_p = ecma_op_to_string (flags_value);
+    ret_value = ecma_op_create_regexp_from_bytecode (new_target_obj_p, bc_p);
+  }
+  else
+  {
+#endif /* ENABLED (JERRY_ES2015) */
 
-    if (JERRY_UNLIKELY (flags_string_p == NULL))
-    {
-      ecma_deref_ecma_string (pattern_string_p);
-      goto cleanup;
-    }
+    ret_value = ecma_op_create_regexp_from_pattern (new_target_obj_p, pattern_value, flags_value);
```

Comment:
I think `pattern_value` and `flags_value` can leak here if `pattern_is_regexp` is true.